app/Actions/AlbumAuthorisationProvider.php

```diff
@@ -280,13 +289,12 @@ public function isAccessible(?AbstractAlbum $album): bool
 
 		if ($album instanceof BaseAlbum) {
 			try {
-				return
-					($album->owner_id === $userID) ||
+				return ($album->owner_id === $userID) ||
 					($album->is_public && $album->password === null) ||
 					($album->is_public && $this->isUnlocked($album)) ||
-					($album->shared_with()->where('user_id', '=', $userID)->count());
+					($album->shared_with()->where('user_id', '=', $userID)->count() > 0);
 			} catch (\InvalidArgumentException $e) {
-				assert(false, new \AssertionError('\InvalidArgumentException must not be thrown by ->where', $e->getCode(), $e));
+				throw new \AssertionError('\InvalidArgumentException must not be thrown by ->where', $e->getCode(), $e);
```

Comment:
After two nights of sleep and input by @ondrejmirtes from PhpStan, I decided to follow @ildyria's idea to write a proper helper method. I decided for an own exception class `LycheeAssertionError` which extends `\AssertionError` and implements `InternalLycheeException`. I fixed this in https://github.com/LycheeOrg/Lychee/pull/1336/commits/f5cd4433a97aef5706f3d661e5c9578e4e7e9938. These were the reasons why I changed my mind:

 1. If the method requires a return type, the PhpStan issue report is accompanied by a second PhpStan issue report `Method <foo> should return <blah> but return statement is missing.` (see [at PhpStan playground](https://phpstan.org/r/2ec8cd91-d089-4859-8b16-457cd36abe2f) for a MWE). Actually, this second error message was the deciding point. If `assert` is optimized away, the `catch`-block itself is not optimized away but still suppresses the original exception. This means the application continuous with its normal program flow and the return value is indeed undefined causing probably much more problems on the outer call stack. What I actually wanted was to optimize away the whole `catch`-block, but of course this is not what happens.
 2. My argument about the efficiency gain was nonsense. Even though `assert` is optimized away if assertions are disabled, the `catch`-block is still there.
 3. By using our own class `LycheeAssertionError` which implements the interface `InternalLycheeException` we can get a grasp on the annoying messages about unhandled and/or uncaught  exceptions. @ondrejmirtes suggested to mark `\InvalidArgumentException` an unchecked exception, but I don't want to go that way, because it marks the exception as unchecked on a global level and there are other places where this exception needs attention.